@here/harp-mapview/lib/MapViewEnvironment.ts

```diff
@@ -47,13 +47,16 @@ export class MapViewEnvironment {
     private m_skyBackground?: SkyBackground;
     private m_createdLights?: THREE.Light[];
     private m_overlayCreatedLights?: THREE.Light[];
+    private readonly m_backgroundDSPromise: Promise<void> | undefined;
     private readonly m_backgroundDataSource?: BackgroundDataSource;
 
     constructor(private readonly m_mapView: MapView, options: MapViewEnvironmentOptions) {
         this.m_fog = new MapViewFog(this.m_mapView.scene);
         if (options.addBackgroundDatasource !== false) {
             this.m_backgroundDataSource = new BackgroundDataSource();
-            this.m_mapView.addDataSource(this.m_backgroundDataSource);
+            this.m_backgroundDSPromise = this.m_mapView.addDataSource(this.m_backgroundDataSource);
```

Comment:
We need to capture this and make it external so that it can be `await`ed in the test to ensure that it is connected before we continue test execution.